### PR TITLE
Create adapter zone docs_generate tests

### LINKED ISSUE #156: [CT-605] Implement adapter zone docs_generate tests in Snowflake
Implement adapter zone docs_generate tests in Snowflake.

### PR DESCRIPTION
**Pull request by gshank**
resolves #156

### Description

Include adapter zone TestDocsGenerateSnowflake

### Checklist

- [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-snowflake next" section.

### CODE REVIEW

**Comment by leahwicz** [comment]
@gshank is this ready for review?

**Comment by McKnight-42** [comment]
@gshank looks like an import error from core for `BaseReference` is that supposed to be `BaseDocsGenReferences`?

**Review by leahwicz — APPROVED** [review]
LGTM. Not sure about the failing CI though...

File: tests/functional/adapter/test_docs_gen_references.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
+import pytest
+from dbt.tests.adapter.basic.test_docs_generate import (
+    BaseReferences, ref_models__ephemeral_copy_sql, ref_models__docs_md
```
Comment (jtcohen6) [inline tests/functional/adapter/test_docs_gen_references.py:3]: It looks like this is named `BaseDocsGenReferences` [in `dbt-tests-adapter`](https://github.com/dbt-labs/dbt-core/blob/189c06dbb16f572d17337a1973aa6768d9abe8cf/tests/adapter/dbt/tests/adapter/basic/test_docs_generate.py#L459)
```suggestion
    BaseDocsGenReferences, ref_models__ephemeral_copy_sql, ref_models__docs_md
```

File: tests/functional/adapter/test_docs_gen_references.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  )
+}}
+
+select first_name, count(*) as CT from {{ref('ephemeral_copy')}}
+group by first_name
+order by first_name asc
+
+"""
+
+class TestReferencesSnowflake(BaseReferences):
```
Comment (jtcohen6) [inline tests/functional/adapter/test_docs_gen_references.py:87]:
```suggestion
class TestReferencesSnowflake(BaseDocsGenReferences):
```

**Pull request closed without merging**